### ISSUE TITLE
Reorganise the build for windows - get rid of zypper.

### ISSUE THREAD

**Issue by oleg68** [GrandOrgue/grandorgue#28]
Now the version for windows is built on an Ubuntu 20.04 GitHub action worker using a cross-compiling technology. The main problem for this is to have all dependences installed on the build machine.

Because Martin Koegler used the OpenSuse build System for all builds, including for Windows, these dependencies are prebuilt as OpenSuse rpm packages. He built two packages: jack and wxWidgets. 

Now Github build uses his packages with zypper. It causes several problems:

1. Zypper is not a native package manager in Ubuntu, so it works not durable and often fails. It causes the failed build.
2. The versions of jack and wxWidgets are quite    old and may work incorrectly with recent Windows versions.
3. Dependency on a stallen repository is not good.

My aproach is to build native ubuntu (debian) packages with windows dependencies. I've already made jack. The next to-do is wxWidgets. Then the prepare-ubuntu20.sh script may be switched from zypper to apt-get.

**Comment by larspalo** [comment]
It's been a long time since I cross-compiled GO to Windows from Linux so I cannot say if something has changed, but I've no memory of ever needing to cross compile jack. I just used the -dev packages as usual and cross-compiled wx myself. The question, if we make our own packages, would of course be what version to use of the libraries. The same stable versions used on Linux, for instance the ones in current LTS release of Ubuntu, or the newest possible releases... Whichever way we go there will inevitably be differences, albeit small, between the releases for different platforms. Anyway, I think it's good not to depend on the old packages, and switch to apt-get.

**Comment by oleg68** [comment]
> The question, if we make our own packages, would of course be what version to use of the libraries. The same stable versions used on Linux, for instance the ones in current LTS release of Ubuntu, or the newest possible releases...

When we build GO for linux, we do not ship dependencies with the GO packages. All rependencies are linked dynamiccaly and are installed with linux package managers during the installation of GO. Usually they choose a newest available version for this distro.  During the build phase we are trying to link GO against as old as possible versions. It increases the compatibility range.

Another thing is building for Windows and OSx. Here we ship all dependencies together with GO, so It is better to use the moct recent version.

**Comment by oleg68** [comment]
> I just used the -dev packages as usual and cross-compiled wx mysel

I've used the build scripts provided by Martin Koegler and I've adopted them for ubuntu. Maybe it is possible to build in another way.

**Comment by larspalo** [comment]
I understand the difference between static and dynamic linking, no issue there. But, we might look into the possibility of providing an appimage for Linux instead of different packages for many distributions. The power users of Linux might frown upon it, but it should make the life easier for any casual user that's moving away from Windows just to test running GO on a Linux computer for any reason. The other option is of course providing a live image as Martin did before... Lets face it, any power user of Linux will know how to compile the program themselves!

I suggest you take the approach that's easiest to maintain. When I've got some time to spare I'll look into this issue more as I've got some interest of getting LoopAuditioneer possible to build in a less manually involved way too.

**Comment by oleg68** [comment]
> But, we might look into the possibility of providing an appimage for Linux instead of different packages for many distributions

We provide only two packages for debian-based and for rpm based. There are some packages for different cpus (amd64, armhf, arm64)

I have already [an issue about appimage](https://github.com/oleg68/GrandOrgue/issues/26). I'm going to transfer all my open issues here after you finish with discussions

**Comment by larspalo** [comment]
Right! I looked at the build script quickly and realize that some (most) of the packages indeed have different names than on Ubuntu. Many of the packages actually don't even exist separately on Ubuntu. I'll try cross compiling on my computer and see if it works out of the box (more or less) here.

**Comment by oleg68** [comment]
The following dependencies are used now:

- mingw64-filesystem
- mingw64-zlib1 mingw64-zlib-devel
- mingw64-libgnurx0 mingw64-libgnurx-devel 
- mingw64-fftw3 mingw64-fftw3-devel
- mingw64-jack mingw64-jack-devel
- mingw64-wavpack mingw64-wavpack-devel 
- mingw64-wxWidgets-3_0-lang mingw64-wxWidgets-3_0-devel
